File: 8_kyu/Expressions_Matter.kt
class Kata {
    companion object {
     
     fun expressionsMatter(a : Int, b : Int, c : Int) : Int =
         arrayOf(a+b+c,
                 a+b*c,
                 (a+b)*c,
                 a*b+c,
                 a*(b+c),
                 a*b*c).max() ?: 0
    }
}

File: 8_kyu/Expressions_Matter.cs
using System.Linq;
using System.Collections.Generic;

public class Kata
{
    public static int ExpressionsMatter(int a, int b, int c)
    {
      IEnumerable<int> allOptions () {
          yield return a+b+c;
          yield return a+b*c;
          yield return (a+b)*c;
          yield return a*b+c;
          yield return a*(b+c);
          yield return a*b*c;
      };
      return allOptions().Max();
    }
}